src/librustdoc/html/static/js/main.js

```diff
@@ -1786,29 +1814,54 @@ href="https://doc.rust-lang.org/${channel}/rustdoc/read-documentation/search.htm
             }
         });
 
-        const el = document.createElement("textarea");
-        el.value = path.join("::");
-        el.setAttribute("readonly", "");
-        // To not make it appear on the screen.
-        el.style.position = "absolute";
-        el.style.left = "-9999px";
-
-        document.body.appendChild(el);
-        el.select();
-        document.execCommand("copy");
-        document.body.removeChild(el);
-
-        but.classList.add("clicked");
+        copyContentToClipboard(path.join("::"));
+        copyButtonAnimation(but);
+    };
 
-        if (reset_button_timeout !== null) {
-            window.clearTimeout(reset_button_timeout);
+    // Copy buttons on code examples.
+    function copyCode(codeElem) {
+        if (!codeElem) {
+            // Should never happen, but the world is a dark and dangerous place.
+            return;
         }
+        let text = "";
+        onEachLazy(codeElem.childNodes, elem => {
+            if (elem.nodeType === Node.TEXT_NODE) {
+                text += elem.textContent;
+            }
+        });
+        copyContentToClipboard(text);
```

Comment:
This is incorrect: It doesn't walk nodes recursively but only shallowly!
Given `let x = 2;` from https://rustdoc.crud.net/imperio/copy-code, it only copies `x = ;` because it skips the `<span>`s instead of recursing into them.

Instead, just replace all of this with `.textContent` which does this traversal for us. Under the hood, it walks the tree and accumulates text nodes.

```suggestion
        copyContentToClipboard(codeElem.textContent);
```

Reply:
https://dom.spec.whatwg.org/#dom-node-textcontent and https://dom.spec.whatwg.org/#get-text-content

Reply:
Much better indeed!

Reply:
Done!